modin/experimental/engines/omnisci_on_ray/frame/calcite_builder.py

```diff
@@ -540,5 +551,7 @@ def _process_sort(self, op):
         self._push(CalciteSortNode(collations))
 
     def _process_filter(self, op):
+        self._maybe_add_projection(op)
```

Comment:
With this projection, we cannot use filters by `rowid` which may be required since it can be used as an index. Applying a projection after the filter is a better option. We should add projection only when the filter is applied to a scan.

Reply:
done